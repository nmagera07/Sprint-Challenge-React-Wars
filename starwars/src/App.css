.App {
  text-align: center;
  
}
.Header {
  color: #443e3e;
  text-shadow: 1px 1px 5px #fff;
}
/* write your parent styles in here for your App.js */
.character {
  display: flex;
  flex-wrap: wrap;
  margin: 200px;
}

.card {
  margin: 25px;
}

img {
  height: 300px;
  margin-bottom: -125px;
  border-radius: 20px;
}