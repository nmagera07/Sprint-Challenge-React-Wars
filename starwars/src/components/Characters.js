import React, {} from 'react';
import { Card, Button, Modal, Icon } from 'semantic-ui-react'

const Characters = (props) => {
    console.log("props",props)
    
    return ( 
        
        <div>
            <div className='character-header'>
            <h1>React Wars</h1>
            <img src="https://vignette.wikia.nocookie.net/starwars/images/c/cc/Star-wars-logo-new-tall.jpg/revision/latest?cb=20190313021755" alt="star wars logo"></img>
            </div>
            <div className="character">
            
            
            {props.character.map(person => 
            <div className="card" key={person.name}>
                
                <Card centered >
                     <Card.Content>
                        <Card.Header>{person.name} <Icon name='space shuttle'/></Card.Header>
                        <Modal trigger={<Button>Click for more info</Button>}>
                            <Modal.Header>{person.name}</Modal.Header>
                            <Modal.Content >
                            <Modal.Description>
                                <p>Birth Year: {person.birth_year}</p>
                                <p>Gender: {person.gender}</p>
                                <p>Height: {person.height}</p>
                                <p>Mass: {person.mass}</p>
                                <p>Eye Color: {person.eye_color}</p>
                            </Modal.Description>
                            </Modal.Content>
                        </Modal>
                        </Card.Content>
                </Card>
            </div>
                )}
                </div>
        </div>
     );
}
 
export default Characters;