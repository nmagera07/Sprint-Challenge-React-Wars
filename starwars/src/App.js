import React, { useEffect, useState} from 'react';
import axios from 'axios'
import './App.css';
import 'semantic-ui-css/semantic.min.css'
import Characters from './components/Characters'
import Nav from './components/Nav'
import DogAPI from './components/DogAPI'
import {BrowserRouter as Router, Route} from 'react-router-dom'

const App = () => {
  // Try to think through what state you'll need for this app before starting. Then build out
  // the state properties here.

  // Fetch characters from the star wars api in an effect hook. Remember, anytime you have a 
  // side effect in a component, you want to think about which state and/or props it should
  // sync up with, if any.
  const [character, setCharacter] = useState([])


  useEffect(() => {
    axios
      .get(`https://swapi.co/api/people/`)
      .then(response => {
        console.log(response)
        setCharacter(response.data.results)
      })
      .catch(error => console.log('things went south...way south'))
  }, [])

  
  return (
    <div className="App">
      <Router>
      <Route path="/" component={Nav} />
      <Route path="/characters" 
        render={props => 
          <Characters {...props} character={character} />
        }
        />
     <Route path="/dogAPI" component={DogAPI}></Route>
      </Router>
      
    </div>
  );
}

export default App;
