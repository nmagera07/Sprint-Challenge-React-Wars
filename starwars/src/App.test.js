import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer'
import Characters from './components/Characters'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('<App />', () => {
  it('should match snapshot', () => {
  const tree = renderer.create(<App />).toJSON()

  expect(tree).toMatchSnapshot()
  })
})

describe('<Characters />', () => {
  it('should match snapshot', () => {
    const tree= renderer.create(<Characters character={ dataArray }/>).toJSON()

    expect(tree).toMatchSnapshot()
  })
})


const dataArray = [{birth_year: "19BBY",
created: "2014-12-09T13:50:51.644000Z",
edited: "2014-12-20T21:17:56.891000Z",
eye_color: "blue",
films: (5) ["https://swapi.co/api/films/2/", "https://swapi.co/api/films/6/", "https://swapi.co/api/films/3/", "https://swapi.co/api/films/1/", "https://swapi.co/api/films/7/"],
gender: "male",
hair_color: "blond",
height: "172",
homeworld: "https://swapi.co/api/planets/1/",
mass: "77",
name: "Luke Skywalker",
skin_color: "fair",
species: ["https://swapi.co/api/species/1/"],
starships: (2) ["https://swapi.co/api/starships/12/", "https://swapi.co/api/starships/22/"],
url: "https://swapi.co/api/people/1/",
vehicles: (2) ["https://swapi.co/api/vehicles/14/", "https://swapi.co/api/vehicles/30/"]}]