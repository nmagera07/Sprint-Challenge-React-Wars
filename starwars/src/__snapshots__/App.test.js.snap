// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div
  className="App"
>
  <div
    className="nav-bar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/characters"
      onClick={[Function]}
    >
      Star Wars API
    </a>
    <a
      href="/dogAPI"
      onClick={[Function]}
    >
      Dog API
    </a>
    <h2>
      Welcome to APIs!
    </h2>
  </div>
</div>
`;

exports[`<Characters /> should match snapshot 1`] = `
<div>
  <div
    className="character-header"
  >
    <h1>
      React Wars
    </h1>
    <img
      alt="star wars logo"
      src="https://vignette.wikia.nocookie.net/starwars/images/c/cc/Star-wars-logo-new-tall.jpg/revision/latest?cb=20190313021755"
    />
  </div>
  <div
    className="character"
  >
    <div
      className="card"
    >
      <div
        className="ui centered card"
        onClick={[Function]}
      >
        <div
          className="content"
        >
          <div
            className="header"
          >
            Luke Skywalker
             
            <i
              aria-hidden="true"
              className="space shuttle icon"
              onClick={[Function]}
            />
          </div>
          <button
            className="ui button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            Click for more info
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
